File: PRACTICA PRIMER PARCIAL/Ejercicio12/Python/GenerarSecuencia.py
def generar_secuencia(n):
    secuencia = [1, 3]
    indice = 1

    for i in range(2, n):
        if i % 2 == 0:
            indice += 1
        else:
            indice += 2

        secuencia.append(indice)

    return secuencia

n_terminos = 20
secuencia = generar_secuencia(n_terminos)
print(secuencia)

File: PRACTICA PRIMER PARCIAL/Ejercicio12/Java/GenerarSecuencia.java
import java.util.ArrayList;
import java.util.List;

public class GenerarSecuencia {

    public static List<Integer> generarSecuencia(int n) {
        List<Integer> secuencia = new ArrayList<>();
        secuencia.add(1);
        secuencia.add(3);
        int indice = 1;

        for (int i = 2; i < n; i++) {
            if (i % 2 == 0) {
                indice++;
            } else {
                indice += 2;
            }

            secuencia.add(indice);
        }

        return secuencia;
    }

    public static void main(String[] args) {
        int nTerminos = 20;
        List<Integer> secuencia = generarSecuencia(nTerminos);

        for (int num : secuencia) {
            System.out.print(num + " ");
        }
    }
}
